File: ruby-gem/lib/models/as_planned_transformation.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #AsPlannedTransformation
require_relative '../core/semantic_object'
require_relative 'transformation'


module DfcLinkmlConnector
  module Models
    class AsPlannedTransformation < Transformation
      SEMANTIC_TYPE = "dfc-b:AsPlannedTransformation".freeze

      # @return [String]
      attr_accessor :input

      # @return [String]
      attr_accessor :output

      # @return [String]
      attr_accessor :transformation_type

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, input: nil, output: nil, transformationType: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, input: nil, output: nil, transformationType: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @input = input
        @output = output
        @transformation_type = transformationType
        self.semanticType = "dfc-b:AsPlannedTransformation"
        registerSemanticProperty("dfc-b:AsPlannedTransformation:has_input", &method("input")).valueSetter = method("input=")
        registerSemanticProperty("dfc-b:AsPlannedTransformation:has_output", &method("output")).valueSetter = method("output=")
        registerSemanticProperty("dfc-b:AsPlannedTransformation:has_transformation_type", &method("transformation_type")).valueSetter = method("transformation_type=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/AsPlannedTransformation.ts
// Class from DFC Business Ontology: #AsPlannedTransformation
import { SemanticObject } from "../core/SemanticObject.js";
import { Transformation, type TransformationParams } from "./Transformation.js";

export interface AsPlannedTransformationParams extends TransformationParams {
  hasInput?: string;
  hasOutput?: string;
  hasTransformationType?: string;
}
export class AsPlannedTransformation extends Transformation {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:AsPlannedTransformation";
  }

  hasInput?: string;
  hasOutput?: string;
  hasTransformationType?: string;

  constructor(
    semanticId: string,
    params?: AsPlannedTransformationParams,
  ) {
    super(semanticId, params);
    this.hasInput = params?.hasInput;
    this.hasOutput = params?.hasOutput;
    this.hasTransformationType = params?.hasTransformationType;
    this.semanticType = AsPlannedTransformation.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:hasInput", () => this.hasInput);
    this.registerSemanticProperty("dfc-b:hasOutput", () => this.hasOutput);
    this.registerSemanticProperty("dfc-b:hasTransformationType", () => this.hasTransformationType);
  }
  static {
    SemanticObject.typeRegistry.set(AsPlannedTransformation.SEMANTIC_TYPE, AsPlannedTransformation);
  }
}
